### PR TITLE
bound hashes inflight and pending fetch size

### LINKED ISSUE #6801: Bound hashes inflight and pending fetch cache
Right now we have a cache that is unbounded:
https://github.com/paradigmxyz/reth/blob/facad24ad64b322b309c2a6547d931cdf0f7bca8/crates/net/network/src/transactions/fetcher.rs#L54

This should be bounded, so it does not grow indefinitely

### PR DESCRIPTION
**Pull request by int88**
fixes: #6801

### CODE REVIEW

File: crates/net/network/src/transactions/fetcher.rs [hunk 1]
```diff
@@ -949,7 +949,9 @@ impl Default for TransactionFetcher {
                 NonZeroUsize::new(DEFAULT_MAX_CAPACITY_CACHE_PENDING_FETCH)
                     .expect("buffered cache limit should be non-zero"),
             ),
-            hashes_fetch_inflight_and_pending_fetch: LruMap::new_unlimited(),
+            hashes_fetch_inflight_and_pending_fetch: LruMap::new(
+                DEFAULT_MAX_CAPACITY_CACHE_INFLIGHT_AND_PENDING_FETCH.try_into().unwrap(),
```
Comment (Rjected) [inline crates/net/network/src/transactions/fetcher.rs:953]: Let's use an `expect` here with a message like the above `expect`

**Review by Rjected — CHANGES_REQUESTED** [review]
one very pedantic nit, otherwise this looks good to me

File: crates/net/network/src/transactions/constants.rs [hunk 2]
```diff
@@ -132,6 +132,15 @@ pub mod tx_fetcher {
     pub const DEFAULT_MAX_CAPACITY_CACHE_PENDING_FETCH: usize =
         100 * SOFT_LIMIT_COUNT_HASHES_IN_GET_POOLED_TRANSACTIONS_REQUEST;
 
+    /// Default max size for cache of inflight and pending transactions fetch.
+    ///
+    /// Default is [`DEFAULT_MAX_CAPACITY_CACHE_PENDING_FETCH`] +
+    /// [`DEFAULT_MAX_COUNT_INFLIGHT_REQUESTS_ON_FETCH_PENDING_HASHES`], which is 25600 hashes and
+    /// 65 requests, so it is 25665.
```
Comment (emhane) [inline crates/net/network/src/transactions/constants.rs:139]:
```suggestion
    /// 65 requests, so it is 25665 hashes.
```

**Review by emhane — APPROVED** [review]
nice!

**Review by Rjected — APPROVED** [review]
looks good, thanks!

**Pull request merged**